packages/core/src/main/conversation/ConversationService.ts

```diff
@@ -144,35 +80,34 @@ export default class ConversationService {
     if (this.shouldSendAsExternal(plainTextBuffer, <UserPreKeyBundleMap>preKeyBundles)) {
       const payload: EncryptedAsset = await AssetCryptography.encryptAsset(plainTextBuffer);
 
-      return this.sendExternalGenericMessage(
-        this.clientID,
-        conversationId,
-        payload,
-        <UserPreKeyBundleMap>preKeyBundles
-      );
+      await this.sendExternalGenericMessage(this.clientID, conversationId, payload, <UserPreKeyBundleMap>preKeyBundles);
+      return messageId;
     }
 
     const payload: OTRRecipients = await this.cryptographyService.encrypt(
       plainTextBuffer,
       <UserPreKeyBundleMap>preKeyBundles
     );
 
-    return this.sendMessage(this.clientID, conversationId, payload);
+    await this.sendMessage(this.clientID, conversationId, payload);
+    return messageId;
   }
 
-  public async sendImage(conversationId: string, image: Image): Promise<ClientMismatch> {
+  public async sendImage(conversationId: string, image: Image): Promise<string> {
     const imageAsset = await this.assetService.uploadImageAsset(image);
+    const messageId = new UUID(4).format();
 
     const genericMessage = this.protocolBuffers.GenericMessage.create({
-      messageId: new UUID(4).format(),
+      messageId,
       asset: imageAsset,
     });
 
     const preKeyBundles = await this.getPreKeyBundles(conversationId);
     const plainTextBuffer: Buffer = this.protocolBuffers.GenericMessage.encode(genericMessage).finish();
     const payload: EncryptedAsset = await AssetCryptography.encryptAsset(plainTextBuffer);
 
-    return this.sendExternalGenericMessage(this.clientID, conversationId, payload, <UserPreKeyBundleMap>preKeyBundles);
+    await this.sendExternalGenericMessage(this.clientID, conversationId, payload, <UserPreKeyBundleMap>preKeyBundles);
```

Comment:
We could return the messageId right away. Here it's waiting for the promise to end before returning it.
Is this the intended behavior?